### PR TITLE
Don't offer completion in a possible deconstruction declaration

### PR DESCRIPTION
**Pull request by jcouv**
The parser treats `var(x, y)` as a method invocation until an equal sign is added, as in `var (x, y) =`.
So tweaking the completion behavior for deconstruction has some negative side-effect on completion in a method invocation where the method is called "var". I think that is ok, because I expect methods called "var" would be quite rare.

With this change, no completions are available, just like there are no completions when typing `var $$`.

Also, I added a test that capture that `($$)` currently only offers soft selections ~~because it is a possible lambda~~.

Also updated the tooltip when in tuple context.

@CyrusNajmabadi @dotnet/roslyn-ide for review.

Fixes https://github.com/dotnet/roslyn/issues/13342

### CODE REVIEW

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb [hunk 1]
```diff
@@ -226,6 +226,75 @@ class C
...
+{
+    public void Foo()
+    {
+        ($$)
+    }
+}]]></Document>)
+
+                state.SendTypeChars("i")
+                Await state.WaitForAsynchronousOperationsAsync()
+                Await state.AssertSelectedCompletionItem(displayText:="int", isSoftSelected:=True)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb]: note: this is not the behavior we actually want.  ```int``` should be hard selected, but typing ```:``` should not commit it.  Can you please comment in this test that this is not desirable behavior, and can you link to the bug about getting hte behavior correct?

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs [hunk 2]
```diff
@@ -469,13 +469,17 @@ public static bool IsConstructorOrMethodParameterArgumentContext(this SyntaxToke
...
                 {
-                    if (targetToken.Parent.IsParentKind(SyntaxKind.InvocationExpression) ||
-                        targetToken.Parent.IsParentKind(SyntaxKind.ObjectCreationExpression) ||
+                    if (targetToken.Parent.IsParentKind(SyntaxKind.ObjectCreationExpression) ||
                         targetToken.Parent.IsParentKind(SyntaxKind.BaseConstructorInitializer) ||
                         targetToken.Parent.IsParentKind(SyntaxKind.ThisConstructorInitializer))
                     {
                         return true;
                     }
+                    if (targetToken.Parent.IsParentKind(SyntaxKind.InvocationExpression) &&
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs]: blank line after }

Also add comment why this condition exists.

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs [hunk 3]
```diff
@@ -1880,6 +1880,17 @@ public static bool IsLabelContext(this SyntaxTree syntaxTree, int position, Canc
                 }
             }
 
+            // var(|
+            // var(id, |
+            if (token.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken))
+            {
+                if (token.Parent.Parent.IsKind(SyntaxKind.InvocationExpression) &&
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: exact same code as above.  This should be extracted out into a helper "IsInvocationOfVarExpression" or something like that.

**Comment by CyrusNajmabadi** [comment]
Minor nits.  If you address everything, then LGTM.

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb (hunk 1 above)
Comment (jcouv) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb]: Filed and linked https://github.com/dotnet/roslyn/issues/13527

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs (hunk 3 above)
Comment (jcouv) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: Factored out.

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs (hunk 2 above)
Comment (jcouv) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs]: Done

**Comment by jcouv** [comment]
I factored a little bigger block into `IsInvocationOfVarExpression`. Let me know if still looks good.

**Comment by CyrusNajmabadi** [comment]
:-1:

**Comment by CyrusNajmabadi** [comment]
er.. i have no idea why there's a thumbs down.  :(

**Comment by CyrusNajmabadi** [comment]
That was supposed to be a :+1:

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs [hunk 4]
```diff
@@ -2174,6 +2183,17 @@ public static bool IsLabelContext(this SyntaxTree syntaxTree, int position, Canc
             return false;
         }
 
+        public static bool IsInvocationOfVarExpression(this SyntaxToken token)
+        {
+            if (token.Parent.Parent.IsKind(SyntaxKind.InvocationExpression) &&
+                ((InvocationExpressionSyntax)token.Parent.Parent).Expression.ToString() == "var")
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: you can just return this boolean expression.

**Pull request merged**